### PR TITLE
Add basic error handling

### PR DESCRIPTION
**Pull request by DavidBM**
I created some structs and set the the returns for starting to delete unwrap. in #19 we have the exit codes we need to have and when to produce them.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -44,7 +61,11 @@ async fn unzip_file(
     #[cfg(unix)]
     set_unix_permissions(&outpath, directory, &file).await;
 
-    Ok(())
+    FileUnzipResult {
```
Comment (elpiel) [inline src/lib.rs]: We can have a different type of error for this instead of having Option for the file.

File: src/lib.rs [hunk 2]
```diff
@@ -1,32 +1,49 @@
...
+
+pub async fn unzip_archive(
+    file: File,
+    directory: &Path,
+) -> Result<Vec<FileUnzipResult>, Box<dyn Error>> {
     let mut archive = zip::ZipArchive::new(file)?;
     let mut results = vec![];
 
     for file_index in 0..archive.len() {
         match archive.by_index(file_index) {
```
Comment (elpiel) [inline src/lib.rs]: We can already handle some of the errors:
http://mvdnes.github.io/rust-docs/zip-rs/zip/result/enum.ZipError.html
I will take a look today, I think we can already handle some of the exit codes!
Note: ZipError is returned in both `by_index`, but also `ZipArchive::new`, so there is some overlap there we need to handle!

File: src/lib.rs (hunk 1 above)
Comment (DavidBM) [reply src/lib.rs]: True! I don't know how much work I will be able to do this week, but feel free to change it :)

Comment (DavidBM) [reply src/lib.rs]: Done!

**Comment by DavidBM** [comment]
I will merge this one

**Pull request merged**